### ISSUE TITLE
linked work items be set to be mandatory for new repositories.

### ISSUE THREAD

**Issue by beforan** [UniversityOfNottingham/universityofnottingham.github.io#10]
following the quality azureance seminar I would like to propose that the linked work items be set to be mandatory for new repositories.

_Originally posted by @Rhino12343 in https://github.com/UniversityOfNottingham/universityofnottingham.github.io/pull/9/files_

**Comment by stylesm** [comment]
I agree, generally. I think there should be something there about exemptions (much like we have for e.g. using Framework), e.g. when creating a repo for an existing codebase that just needs adding to source control and a quick bugfix doing - some legacy app that only gets touched once every 2 years. Alternatively, have a catch-all project or team board on DevOps for the small changes (I think one did sort of exist, has some Bluecastle/Legend stuff in it)?

**Comment by beforan** [comment]
catch-all board is an option, though may be unpopular

**Comment by stylesm** [comment]
Hmm why is that? If linked work items is going to be enforced then the alternative is mandating a team for each software project. Possible, but also seems it would be unpopular given the faff just to create a single work item.

Thinking of things like insurance renewals which gets a couple of bugs a year, for example.

**Comment by Rhino12343** [comment]
I would like to see something along the lines of a project board for active development, a separate board for catch all bug fixes (with no active development), and a board for small feature requests.

**Comment by stylesm** [comment]
Per project?

**Comment by Rhino12343** [comment]
Per project for active development.
Catch all board for bug fixes on non active projects.
Catch all board for feature requests on non active projects.

**Comment by beforan** [comment]
So there is some thinking that a catch-all board for bugs on all non active projects makes developers assigned to that work very sad about the world.

**Comment by stylesm** [comment]
Then presumably the alternative (if this standard is to be adopted) is to have a team/board for each project regardless of how small the development is; if a board doesn't exist when a bug/feature needs doing, one should be created?

Be interested to know what other people would want - would they just not want to track it on DevOps (meaning this standard couldn't be enforced or would require a paragraph about exemptions)? I can understand that logic if all you're doing is changing some text on Absence Reporting.

Though my question would be - do we have developers who aren't using DevOps at all and/or PMO are still emailing copies of spreadsheets around? At what point does a 'small piece of work' require tasking out on DevOps?

**Comment by stylesm** [comment]
As an aside, if we are to have a catch-all board/team, I would prefer a single one for bugs/features, rather than separate. I think if you were asked to work on SPMIC (which has a list of outstanding bugs and features) for example, having two teams/boards could get faffy, personally.

**Comment by beforan** [comment]
> would they just not want to track it on DevOps (meaning this standard couldn't be enforced or would require a paragraph about exemptions)

that appears to be the case in practice right now.

**Comment by stylesm** [comment]
> > would they just not want to track it on DevOps (meaning this standard couldn't be enforced or would require a paragraph about exemptions)
> 
> that appears to be the case in practice right now.

But does that apply to just single/couple of bug fixes (e.g. tickets in SupportWorks), or small 'programmes of work', e.g. thetrainline Workspace component, or the SSIS integration for the HR case management system?

The former sounds like a reasonable exception.

**Comment by beforan** [comment]
the former does sound like a reasonable exception.

I can think of two pieces of integration work right now - one standing up an api that could (should) become reusable, and one that's a console app for some ETL - that aren't being board managed.

**Comment by stylesm** [comment]
Ok, so this standard might help with adopting use of DevOps (particularly with PMO if we can say well it's the standards)..

**Comment by beforan** [comment]
that is part of the intent of it, for sure
